CMakeLists.txt

```diff
@@ -20,7 +19,10 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
+file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/include/hypercomm/core)
+file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/include/hypercomm/messaging)
+
 add_subdirectory(src)
 
 # Generate template for tests
-configure_file(Makefile.common.in Makefile.common)
+configure_file(Makefile.common.in ${PROJECT_SOURCE_DIR}/Makefile.common)
```

Comment:
This should be `CMAKE_SOURCE_DIR`. `PROJECT_SOURCE_DIR` is set to the directory where CMake encountered last call to `project(...)`.